src/Common/src/Interop/Windows/NtDll/Interop.NtQueryInformationProcess.cs

```diff
@@ -0,0 +1,58 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.Win32.SafeHandles;
+using System;
+using System.Runtime.InteropServices;
+
+internal partial class Interop
+{
+    internal partial class NtDll
+    {
+        [DllImport(Libraries.NtDll, CharSet = CharSet.Unicode)]
+        internal static extern int NtQueryInformationProcess(SafeProcessHandle hProcess, PROCESSINFOCLASS pic, ref PROCESS_BASIC_INFORMATION pbi, int cb, ref int pSize);
+		
+		internal enum PROCESSINFOCLASS : int
+        {
+            ProcessBasicInformation = 0,
+            ProcessQuotaLimits,
+            ProcessIoCounters,
+            ProcessVmCounters,
+            ProcessTimes,
+            ProcessBasePriority,
+            ProcessRaisePriority,
+            ProcessDebugPort,
+            ProcessExceptionPort,
+            ProcessAccessToken,
+            ProcessLdtInformation,
+            ProcessLdtSize,
+            ProcessDefaultHardErrorMode,
+            ProcessIoPortHandlers, // Note: this is kernel mode only
+            ProcessPooledUsageAndLimits,
+            ProcessWorkingSetWatch,
+            ProcessUserModeIOPL,
+            ProcessEnableAlignmentFaultFixup,
+            ProcessPriorityClass,
+            ProcessWx86Information,
+            ProcessHandleCount,
+            ProcessAffinityMask,
+            ProcessPriorityBoost,
+            MaxProcessInfoClass
+        };
+
+        [StructLayout(LayoutKind.Sequential)]
+        internal struct PROCESS_BASIC_INFORMATION
+        {
+            public IntPtr ExitStatus;
```

Comment:
Most of these aren't pointers, but LONG/ULONG under the covers, which I believe are always 32 bit, whereas IntPtr is either 32 or 64 bit.
Seems like it should be `int`, `IntPtr`, `uint`, `int`, `uint`, `uint` ? based on the Windows headers.

Reply:
@JeremyKuhne did I get this right? if so I don't know why it does'nt crash as it is here.

Reply:
If it helps in sorting through this: I copied the struct field definitions from https://github.com/Microsoft/msbuild/blob/0591c15d6c638cad38091fbe625dde968f86748d/src/Shared/NativeMethodsShared.cs#L304-L309

Reply:
Note: If we end up sticking with the IntPtr style, `IntPtr UniqueProcessId` should be changed to a `UIntPtr` (assuming that  https://www.pinvoke.net/default.aspx/Structures/PROCESS_BASIC_INFORMATION.html is correct).

Reply:
The MSBuild def is wrong unfortunately. Here is the correct definition:

``` C++
//
// Basic and Extended Basic Process Information
//  NtQueryInformationProcess using ProcessBasicInformation
//

typedef struct _PROCESS_BASIC_INFORMATION {
    NTSTATUS ExitStatus;
    PPEB PebBaseAddress;
    ULONG_PTR AffinityMask;
    KPRIORITY BasePriority;
    ULONG_PTR UniqueProcessId;
    ULONG_PTR InheritedFromUniqueProcessId;
} PROCESS_BASIC_INFORMATION,*PPROCESS_BASIC_INFORMATION;
```

``` C#
// See ntddk.h / winternl.h
internal struct PROCESS_BASIC_INFORMATION
{
    public uint ExitStatus;
    public IntPtr PebBaseAddress;
    public UIntPtr AffinityMask;
    public int BasePriority;
    public UIntPtr UniqueProcessId;
    public UIntPtr InheritedFromUniqueProcessId;
}
```

This header may only show up if you install the WDK- I have it installed on all my machines. If you can't find either of the files under `C:\Program Files (x86)\Windows Kits`, install the WDK and you'll have the files there.

Reply:
Go with what @JeremyKuhne has 

I had mapped NTSTATUS to int since it's a LONG not a ULONG. I guess that's just a historical quirk on their side. For ULONG_PTR I read `typedef ULONG ULONG_PTR;` but didn't notice that was bitness specific so it should indeed by UIntPtr not uint.

Reply:
BTW you might want to open an issue against MSBuild to fix their def. Presumably they've got away with it so far due to the way memory happens to be laid out and the function happens to work.

Reply:
[msbuild issue created](https://github.com/Microsoft/msbuild/issues/3638).